src/client/tcp_dns.js

```diff
@@ -1,37 +1,50 @@
 const net = require('net')
 const dns = require('dns')
+const util = require('util')
+const resolveSrv = util.promisify(dns.resolveSrv)
 
-const debug = require('debug')('minecraft-protocol')
-
-module.exports = function (client, options) {
+module.exports = (client, options) => {
   // Default options
   options.port = options.port || 25565
   options.host = options.host || 'localhost'
+  if (options.connect) return
+  options.connect = async (passedClient) => connect(passedClient, options)
+}
 
-  if (!options.connect) {
-    options.connect = (client) => {
-      // Use stream if provided
-      if (options.stream) {
-        client.setSocket(options.stream)
-        return client.emit('connect')
-      }
-
-      // If port was not defined (defaults to 25565), host is not an ip and is not localhost
-      if (options.port === 25565 && net.isIP(options.host) === 0 && options.host !== 'localhost') {
-        // Try to resolve SRV records for the comain
-        dns.resolveSrv(`_minecraft._tcp.${options.host}`, (err, addresses) => {
-          // Error resolving domain
-          if (err) debug(`srv lookup failed: ${err}`)
+async function connect (client, options) {
+  // Use stream if provided
+  if (options.stream) {
+    client.setSocket(options.stream)
+    client.emit('connect')
+    return
+  }
 
-          // SRV Lookup resolved correctly
-          if (addresses && addresses.length > 0) {
-            debug(`srv lookup returned ${addresses}`)
-            options.host = addresses[0].name
-            options.port = addresses[0].port
-          }
-        })
+  // If port was not defined (defauls to 25565), host is not an ip neither localhost
+  if (
+    options.port === 25565 &&
+        net.isIP(options.host) === 0 &&
+        options.host !== 'localhost'
+  ) {
+    // Try to resolve SRV records for the comain
+    try {
+      const addresses = await resolveSrv(`_minecraft._tcp.${options.host}`)
+      const [address] = addresses
+      const { name, port } = address
```

Comment:
If address is undefined, the destructuration will error.

Reply:
That's true, but the catch would handle it, I tried entering an arbitrary invalid address, `dhusah.afsasd`, and it was caught by the catch which then attempts to call the error resolving domain, which doesn't work, so you get the error described below. I am also adding debugging however that will tell you when srv lookup fails. This is the same outcome that happened before my pr though.
```js
Error: getaddrinfo ENOTFOUND dhusah.afsasd
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26)
    at GetAddrInfoReqWrap.callbackTrampoline (internal/async_hooks.js:129:14) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'dhusah.afsasd'
}
```